src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java

```diff
@@ -0,0 +1,186 @@
+ */
+package org.hawkular.client.android.auth;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.UUID;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.jboss.aerogear.android.authorization.AuthzModule;
+import org.jboss.aerogear.android.authorization.oauth2.OAuth2AuthzSession;
+import org.jboss.aerogear.android.core.Callback;
+import org.jboss.aerogear.android.pipe.http.HeaderAndBody;
+import org.jboss.aerogear.android.pipe.http.HttpException;
+import org.jboss.aerogear.android.pipe.http.HttpProvider;
+import org.jboss.aerogear.android.pipe.http.HttpRestProvider;
+import org.jboss.aerogear.android.pipe.module.AuthorizationFields;
+import org.jboss.aerogear.android.pipe.module.ModuleFields;
+import org.jboss.aerogear.android.store.DataManager;
+import org.jboss.aerogear.android.store.generator.IdGenerator;
+import org.jboss.aerogear.android.store.sql.SQLStore;
+import org.jboss.aerogear.android.store.sql.SQLStoreConfiguration;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Base64;
+import timber.log.Timber;
+
+/**
+ * ModifiedAuthzModule.
+ * <p>
+ * Performs all related work of generating new key and secret pair, Storing them as account,
+ * Retrieving stored account, Deleting stored account.
+ */
+
+public class SecretStoreAuthzModule implements AuthzModule {
+    private final Context context;
+    private final SQLStore<OAuth2AuthzSession> sessionStore;
+
+    public SecretStoreAuthzModule(Context context) {
+        this.context = context.getApplicationContext();
+        this.sessionStore = openSessionStore();
+        sessionStore.openSync();
+    }
+
+    private SQLStore<OAuth2AuthzSession> openSessionStore() {
+        DataManager.config(AuthData.STORE, SQLStoreConfiguration.class)
+                .forClass(OAuth2AuthzSession.class)
+                .withContext(context)
+                .withIdGenerator(new IdGenerator() {
+                    @Override
+                    public String generate() {
+                        return UUID.randomUUID().toString();
+                    }
+                }).store();
+        return (SQLStore<OAuth2AuthzSession>) DataManager.getStore(AuthData.STORE);
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        return true;
+    }
+
+    @Override
+    public boolean hasCredentials() {
+        return sessionStore.read(AuthData.NAME) != null;
+    }
+
+    public void requestAccess(final Callback<String> callback,
+                              final String username, final String password, final URL url) {
+
+        AsyncTask<Void, Void, HeaderAndBody> task = new AsyncTask<Void, Void, HeaderAndBody>() {
+            private Exception exception;
+
+            @Override
+            protected HeaderAndBody doInBackground(Void... params) {
+                HeaderAndBody result;
+                try {
+                    HttpProvider provider =
```

Comment:
Just curious - why do we here use a HttpProvider and not one of the pipes? (bear with me if I am confusing things)